public/app/features/alerting/unified/hooks/useFolder.ts

```diff
@@ -1,35 +1,31 @@
 import { useEffect } from 'react';
 
-import { FolderDTO, useDispatch } from 'app/types';
-
-import { fetchFolderIfNotFetchedAction } from '../state/actions';
-import { initialAsyncRequestState } from '../utils/redux';
-
-import { useUnifiedAlertingSelector } from './useUnifiedAlertingSelector';
+import { useLazyGetFolderQuery } from 'app/features/browse-dashboards/api/browseDashboardsAPI';
+import { FolderDTO } from 'app/types';
 
 interface ReturnBag {
   folder?: FolderDTO;
   loading: boolean;
 }
 
+const PREFER_CACHED_VALUES = true;
+
+/**
+ * Returns a folderDTO for the given uid – uses cached values
+ * @TODO propagate error state
+ */
 export function useFolder(uid?: string): ReturnBag {
-  const dispatch = useDispatch();
-  const folderRequests = useUnifiedAlertingSelector((state) => state.folders);
+  const [fetchFolder, fetchFolderState] = useLazyGetFolderQuery();
```

Comment:
Any reason to use a lazy query instead of `useGetFolderQueryHook`?
If there is no `uid` we could skip the request

Reply:
Without the lazy one we'd have to resort coercing the type checker (with `id!`) and keep that in sync with the skip parameter

Reply:
We can try to use `skipToken` https://redux-toolkit.js.org/rtk-query/usage-with-typescript#skipping-queries-with-typescript-using-skiptoken